app/models/manageiq/providers/redhat/infra_manager/vm.rb

```diff
@@ -48,17 +48,18 @@ def parent_cluster
   alias ems_cluster parent_cluster
 
   def disconnect_storage(_s = nil)
-    unless active?
-      return
-    end
+    return unless active?
+
+    vm_storages = ([storage] + storages).compact.uniq
+    return if vm_storages.empty?
+
     vm_disks = collect_disks
 
-    if vm_disks.blank?
-      storage = nil
-    else
-      vm_storages = ([storage] + storages).compact.uniq
-      storage = vm_storages.select { |store| !vm_disks.include?(store.ems_ref) }
-    end
+    storage = if vm_disks.blank?
```

Comment:
I think this will work and looks more compact:
`storage = vm_storages.select { |store| !vm_disks.include?(store.ems_ref) } if vm_disks.blank?`